src/message-builder.js

```diff
@@ -21,16 +21,61 @@ function convertChannelNameToId(channel, channels) {
     }
     return null;
 }
+//* Repeats weekly (+ 7 ). 
+function incrimentDateIterative(startDate, endDate) {
+    
+    startDate = new Date(startDate) 
+    endDate = new Date(endDate)
+    let dateList = []
+    let date
+   
+    while (startDate !== endDate && startDate < endDate) {     
+      date = new Date(startDate.setDate(startDate.getDate() + 7));      
```

Comment:
I did this because setDate() updates the date object, so in each iteration the startDate changes-> using getDate() to add 7 to the updated Date object, and eventually finishing the while loop with the updated Date.

Reply:
I would write it diferently, since with your usage its not ovbious what is happening here. I would suggest the following:

Instead of:
```
function incrimentDateIterative(startDate, endDate) {
    
    startDate = new Date(startDate) 
    endDate = new Date(endDate)
    let dateList = []
    let date
   
    while (startDate !== endDate && startDate < endDate) {     
      date = new Date(startDate.setDate(startDate.getDate() + 7));      
      dateList.push(date)
        
    } 
    return dateList;
  }
```

This makes it more clear:
```
function incrimentDate(startDate, endDate) {
    
    startDate = new Date(startDate) ;
    endDate = new Date(endDate);
    let dateList = [] ;
    let date = startDate;
   
    while (date !== endDate && date < endDate) {     
     
      date = new Date(date.getDate() + 7);       //Not sure if we also need the getDate here?
      dateList.push(date)
        
    } 
    return dateList;
  }
```

Reply:
oooo love that edit, so simple, so clean! will test it out

Reply:
Oh and 
```
while (date !== endDate && date < endDate)
```
Should also be just
```
while (date < endDate)
```
Since the second condition always includes the first condition.

Reply:
lol at me for missing that logic...
but yea still need the setDate though, to update the date object, it will continue in an infinite loop without. The Date obj has some behaviors I've not seen before.. Im not sure why assigning the 'date' to new object would not solve the issue. But, it does continue into an infinite loop without it.